src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -163,10 +185,12 @@ public static GroupTag parseGroupTag(String tag) throws ParseException {
     public static ModuleTag parseModuleTag(String tag) throws ParseException {
         requireNonNull(tag);
         String trimmedTag = tag.trim();
-        if (!ModuleTag.isValidTagName(trimmedTag)) {
+        String[] args = parseModuleMore(trimmedTag);
+        if (!ModuleTag.isValidTagName(args[0])) {
             throw new ParseException(ModuleTag.MESSAGE_CONSTRAINTS);
         }
-        return new ModuleTag(trimmedTag);
+        // need assertions for other arguments as well.
```

Comment:
Problem here is that there is a lack of input sanitization.

Reply:
`requireNonNull()` works too

Reply:
If you’re worried about sanitation, you could do pattern matching. Refer to EduMateParser and how it checks the overarching pattern of the command.

Reply:
Also, Optional works wonders for this type of thing.